### PR TITLE
Update  Kotlin Convention

### PR DESCRIPTION
**Pull request by at-hoavo**

### CODE REVIEW

**Comment by at-hoavo** [comment]
please review for me, thank you!

File: draft/Basic.md [hunk 1]
```diff
@@ -176,4 +191,73 @@ data class Test(var name: String) {
         this@Test.name = name
     }
 }
+~~~
+## Where to break
```
Comment (at-vinhhuynh) [inline draft/Basic.md]: Please add a break line at start and end of this line

File: draft/Basic.md [hunk 2]
```diff
@@ -176,4 +191,73 @@ data class Test(var name: String) {
...
+Code has a column limit of **100** characters. Any line that would exceed this limit must be line-wrapped
+   
+   * When a line is broken at an `assignment operator( =, +=, -=, *=, /=, %=)` the break comes after the symbol.
+
+~~~kotlin
+fun compare(a: String, b: String): Boolean =
+            // ...
+~~~
```
Comment (at-vinhhuynh) [inline draft/Basic.md]: please add a break line at below

File: draft/Basic.md [hunk 3]
```diff
@@ -176,4 +191,73 @@ data class Test(var name: String) {
...
+* The break comes before the symbol `.` and `::` and `non-assignment operator`
+   
+~~~kotlin
```
Comment (at-vinhhuynh) [inline draft/Basic.md]: Please add break line at below

File: draft/Basic.md [hunk 4]
```diff
@@ -176,4 +191,73 @@ data class Test(var name: String) {
...
+addMarker(MarkerOptions()
+                    .position(currentLocation)
+                    .draggable(true)
+                    .title(resources.getString(R.string.current_location))
+~~~
```
Comment (at-vinhhuynh) [inline draft/Basic.md]: Please add break line at below

File: draft/Basic.md [hunk 5]
```diff
@@ -176,4 +191,73 @@ data class Test(var name: String) {
...
+fun getSomething(
+        a:String,
+        b:String,
+        c:String,
+        d:String,
+        ::isSomething
+){
+ // Do something..
+}
+~~~
```
Comment (at-vinhhuynh) [inline draft/Basic.md]: Please add break line at below

File: draft/Basic.md [hunk 6]
```diff
@@ -176,4 +191,73 @@ data class Test(var name: String) {
...
+   
+~~~kotlin
+fun <T> Iterable<T>.joinToString(
+        separator: CharSequence = ", ",
+        prefix: CharSequence = "",
+        postfix: CharSequence = ""
+): String {
+    // …
+}     
+~~~
```
Comment (at-vinhhuynh) [inline draft/Basic.md]: Please add break line at below

File: draft/Basic.md [hunk 7]
```diff
@@ -176,4 +191,73 @@ data class Test(var name: String) {
...
+   * A lambda arrow (`->`) stays attached to the argument list what precedes it.
+   
+~~~kotlin
+val printSummary = { username: String, age: String, score: Int ->
+            println("User '$username' with '$age' get $score points.")
+        }
+~~~
```
Comment (at-vinhhuynh) [inline draft/Basic.md]: Please add break line at below

File: draft/Basic.md [hunk 8]
```diff
@@ -176,4 +191,73 @@ data class Test(var name: String) {
...
+## Annotation
+   * Member or type annotations are placed on separate lines immediately prior to the annotated construct.
+   
+~~~kotlin
+@Retention(SOURCE)
+@Target(FUNCTION, PROPERTY_SETTER, FIELD)
+annotation class Global
+~~~
```
Comment (at-vinhhuynh) [inline draft/Basic.md]: Please add break line at below

File: draft/Basic.md [hunk 9]
```diff
@@ -176,4 +191,73 @@ data class Test(var name: String) {
...
+   * Annotations without arguments can be placed on a single line.
+   
+~~~kotlin
+@JvmField @Volatile
+var disposable: Disposable? = null
+~~~
```
Comment (at-vinhhuynh) [inline draft/Basic.md]: Please add break line at below

File: draft/Java_Convert_Kotlin.md [hunk 10]
```diff
@@ -26,6 +26,66 @@ vn.asiantech.project
 | Dimensions | `res/values/dimens.xml` | This is where we put [dimension values](https://developer.android.com/guide/topics/resources/more-resources.html). 
 | String | `res/values/strings.xml` | This is where we put strings.
 | Styles |` res/values/styles.xml`| This is where we put style values.
+### •	File structure
+A .kt file comprises of the following, in order:
+~~~kotlin
```
Comment (at-vinhhuynh) [inline draft/Java_Convert_Kotlin.md]: Please add break line at above

File: draft/Java_Convert_Kotlin.md [hunk 11]
```diff
@@ -26,6 +26,66 @@ vn.asiantech.project
...
+Copyright and/or license header (optional)
+Package statement
+Import statements
+Top-level declarations
+~~~
+   * ##### Copyright and/or license header (optional)
+   If a **copyright or license header** belongs in the file it should be placed at the immediate top in a **multi-line comment**
```
Comment (at-vinhhuynh) [inline draft/Java_Convert_Kotlin.md]: Please add break line at below

**Comment by at-hoavo** [comment]
@at-vinhhuynh please review for me, thank you!

**Pull request merged**